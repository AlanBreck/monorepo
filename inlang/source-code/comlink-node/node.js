import { Worker as NodeWorker } from "node:worker_threads"
import nodeAdapter from "comlink/dist/esm/node-adapter.mjs"

export const WorkerPrototype = NodeWorker
export const adapter = nodeAdapter
