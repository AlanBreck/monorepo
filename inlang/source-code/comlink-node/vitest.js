// eslint-disable-next-line no-undef
export const WorkerPrototype = globalThis.Worker
export const adapter = (x) => x
