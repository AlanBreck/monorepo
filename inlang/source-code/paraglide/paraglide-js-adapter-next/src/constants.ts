export enum HeaderNames {
	Link = "Link",
	ParaglideLanguage = "x-language-tag",
	AcceptLanguage = "accept-language",
}
