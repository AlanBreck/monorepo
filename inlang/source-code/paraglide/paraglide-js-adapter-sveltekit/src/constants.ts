/**
 * The Alias for the ParaglideJS Outdir
 * The vite plugin will replace this with the actual outdir
 *
 * This allows you to write imports like this:
 * @example
 * ```ts
 * import { languageTag } from {OUTDIR_ALIAS}/runtime.js"
 * ```
 */
export const OUTDIR_ALIAS = "paraglide-adapter-sveltekit:outdir:"
export const TRANSLATE_PATH_MODULE_ID = "paraglide-adapter-sveltekit:translate-path"
export const GET_LANGUAGE_MODULE_ID = "paraglide-adapter-sveltekit:get-language"
export const HEADER_COMPONENT_MODULE_ID = "paraglide-adapter-sveltekit:header-component.svelte"
export const HEADER_COMPONENT_NAME = "PARAGLIDE_ADAPTER_SVELTEKIT_HEADER"

//The virtual module ID for the public interface
export const PUBLIC_VIRTUAL_MODULE_ID = "$paraglide-adapter-sveltekit"

/**
 * The name to give the `languageTag` function when injecting it into the code
 */
export const LANGUAGE_TAG_ALIAS = "paraglide_adapter_sveltekit_language_tag"
export const TRANSLATE_PATH_FUNCTION_NAME = "paraglide_adapter_sveltekit_translate_path"
