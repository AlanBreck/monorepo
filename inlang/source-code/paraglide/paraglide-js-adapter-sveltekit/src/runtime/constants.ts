export const PARAGLIDE_CONTEXT_KEY = "$paraglide-adapter-sveltekit:context"

/** The path suffix SvelteKit adds on Data requests */
export const DATA_SUFFIX = "__data.json"
export const HTML_DATA_SUFFIX = ".html__data.json"