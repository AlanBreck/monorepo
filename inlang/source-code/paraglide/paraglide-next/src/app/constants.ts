export const LINK_HEADER_NAME = "Link"
export const PARAGLIDE_LANGUAGE_HEADER_NAME = "x-language-tag"
export const ACCEPT_LANGUAGE_HEADER_NAME = "accept-language"

export const LANG_COOKIE_NAME = "NEXT_LOCALE"
