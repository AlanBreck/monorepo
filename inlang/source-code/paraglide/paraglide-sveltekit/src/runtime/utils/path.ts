/**
 * Always starts with a slash and never ends with a slash.
 */
export function normalize(path: string): `/${string}` {
	return `/${path.split("/").filter(Boolean).join("/")}`
}

/**
 * Join path segments
 * 
 * @example
 * ```ts
 * resolve(["some","sub/page"]) === "/some/sub/page"
 * ```
 */
export const resolve = (...segments: string[]) =>
	segments
		.map(normalize)
		.filter((s) => s !== "/")
		.reduce((a, b) => a + b, "") || "/"
