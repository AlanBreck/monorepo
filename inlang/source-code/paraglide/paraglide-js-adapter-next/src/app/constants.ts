import { CookieConfig } from "./utils/cookie"

export const HeaderNames = {
	Link: "Link",
	ParaglideLanguage: "x-language-tag",
	AcceptLanguage: "accept-language",
} as const

// settings for the native language cookie in the pages router
// someone on discord suggested we use these, which seems reasonable
export const LANG_COOKIE: Omit<CookieConfig, "value"> = {
	name: "NEXT_LOCALE",
	maxAge: 31557600, //Math.round(60 * 60 * 24 * 365.25) - 1 year,
	sameSite: "lax",
} as const
