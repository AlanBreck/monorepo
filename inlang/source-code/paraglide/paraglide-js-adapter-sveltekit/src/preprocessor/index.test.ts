import { describe, it, expect } from "vitest"
import { preprocess } from "./index.js"
import { LANGUAGE_TAG_ALIAS } from "../constants.js"
import glob from "tiny-glob"
import fs from "node:fs/promises"
import { basename, resolve } from "node:path"

const testFiles = await glob("./src/preprocessor/fixtures/*.svelte")
console.log(testFiles)

const preprocessor = preprocess()

describe("preprocessor", () => {
	for (const file of testFiles) {
		const filename = basename(file)
		const testName = filename.replace(".svelte", "").replace(/-/g, " ")

		it(testName, async () => {
			const filePath = resolve(process.cwd(), file)
			const content = await fs.readFile(filePath, "utf-8")

			const result = preprocessor.markup({ content, filename: filePath })
			expect(result.code).toMatchSnapshot()
		})
	}
})