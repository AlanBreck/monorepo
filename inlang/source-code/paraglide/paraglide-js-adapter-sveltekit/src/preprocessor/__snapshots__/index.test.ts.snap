// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`preprocessor > interpolation 1`] = `
"<script>
import { languageTag as paraglide_adapter_sveltekit_language_tag } from 'paraglide-adapter-sveltekit:outdir:/runtime.js';

    let name = \\"Loris\\";
</script>
<a href={translatePath(\`/hello-\${name}\`, paraglide_adapter_sveltekit_language_tag())}>Hello {name}</a>"
`;

exports[`preprocessor > mutliple hrefs 1`] = `
"<script>
import { languageTag as paraglide_adapter_sveltekit_language_tag } from 'paraglide-adapter-sveltekit:outdir:/runtime.js';
</script>
<h1>
	<a href={translatePath(\`https://inlang.com\`, paraglide_adapter_sveltekit_language_tag())}>inlang</a>
	<a href={translatePath(\`https://inlang.com\`, paraglide_adapter_sveltekit_language_tag())}>inlang</a>
	<a href={translatePath(\`https://inlang.com\`, paraglide_adapter_sveltekit_language_tag())}>inlang</a>
</h1>
"
`;

exports[`preprocessor > text href quoted 1`] = `
"<script>
import { languageTag as paraglide_adapter_sveltekit_language_tag } from 'paraglide-adapter-sveltekit:outdir:/runtime.js';
</script>
<a href={translatePath(\`/about\`, paraglide_adapter_sveltekit_language_tag())}>About</a>"
`;

exports[`preprocessor > text href unquoted 1`] = `
"<script>
import { languageTag as paraglide_adapter_sveltekit_language_tag } from 'paraglide-adapter-sveltekit:outdir:/runtime.js';
</script>
<a href={translatePath(\`/about\`, paraglide_adapter_sveltekit_language_tag())}>About</a>"
`;
